File: src/main/java/corp/soyummy/userfavorites/UserFavoriteRepository.java
package corp.soyummy.userfavorites;

import org.springframework.stereotype.Repository;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.bson.types.ObjectId;
import java.util.List;

@Repository
public interface UserFavoriteRepository extends MongoRepository<UserFavorite, String> {
    List<UserFavorite> findByUserId(ObjectId userId);
    List<UserFavorite> findByRecipeId(ObjectId recipeId);
}

File: src/main/kotlin/com/soyummy/userfavorites/UserFavoriteRepository.kt
package com.soyummy.userfavorites

import org.springframework.stereotype.Repository
import org.springframework.data.mongodb.repository.MongoRepository
import org.bson.types.ObjectId

@Repository
interface UserFavoriteRepository: MongoRepository<UserFavorite, String> {
    fun findByUserId(userId: ObjectId): List<UserFavorite>
    fun findByRecipeId(recipeId: ObjectId): List<UserFavorite>
}